lemmy_api/src/user.rs

```diff
@@ -347,6 +348,19 @@ impl Perform for SaveUserSettings {
     let preferred_username = diesel_option_overwrite(&data.preferred_username);
     let matrix_user_id = diesel_option_overwrite(&data.matrix_user_id);
 
+    // Check to make sure the avatar and banners are urls
+    if let Some(Some(avatar)) = &avatar {
+      if Url::parse(avatar).is_err() {
+        return Err(APIError::err("invalid_url").into());
+      }
+    }
```

Comment:
Tbh I would just make a quick helper method for this.

Reply:
Good call, I'll do this.